File: server/services/mailgun.py
from dataclasses import dataclass

import httpx


@dataclass
class From:
    email: str
    name: str


class MailgunClient:
    def __init__(self, api_key: str, domain_name: str):
        self.api_key = api_key
        self.domain_name = domain_name

    def send_email(self,
                   from_: From,
                   to: str,
                   subject: str,
                   html_content: str,
                   ):
        return httpx.post(
            f"https://api.mailgun.net/v3/{self.domain_name}/messages",
            auth=("api", self.api_key),
            data={"from": f"{from_.name} <{from_.email}>",
                  "to": to,
                  "subject": subject,
                  "html": html_content,
                  })

File: server/services/mailgun.ts
export class MailgunService implements MailService {
  constructor(
    private mailgunApiKey: string,
    private mailgunDomainName: string,
  ) {
  }

  async sendEmail(
    from: { email: string, name: string },
    to: string,
    subject: string,
    html: string
  ) {
    const form = new FormData()
    form.append("from", `${from.name} <${from.email}>`)
    form.append("to", to)
    form.append("subject", subject)
    form.append("html", html)

    return await $fetch.raw(`https://api.mailgun.net/v3/${this.mailgunDomainName}/messages`, {
      method: "POST",
      headers: {
        Authorization: "Basic " + Buffer.from(`api:${this.mailgunApiKey}`).toString("base64"),
      },
      body: form
    })
  }
}
